### PR TITLE
Mania judgements

### PR DESCRIPTION
**Pull request by smoogipoo**
I'm sort of stabbing in the dark here and seeing how things turn out with a new algorithm separate to osu-stable. This needs more proper experimentation before rankings happen, probably founded by data based on average drain rates, etc...

Completes https://github.com/ppy/osu/issues/709

### CODE REVIEW

File: osu.Game.Rulesets.Mania/Scoring/ManiaScoreProcessor.cs [hunk 1]
```diff
@@ -19,15 +160,132 @@ public ManiaScoreProcessor(HitRenderer<ManiaHitObject, ManiaJudgement> hitRender
...
+                    {
+                        // Head
+                        AddJudgement(new ManiaJudgement
+                        {
+                            Result = HitResult.Hit,
+                            ManiaResult = ManiaJudgement.MAX_HIT_RESULT
+                        });
+
+                        // Ticks
+                        for (int i = 0; i < ((HoldNote)obj).TotalTicks; i++)
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/Scoring/ManiaScoreProcessor.cs]: this count will evaluate every loop. i'd remove `TotalTicks` and just figure it out locally here.

Comment (smoogipooo) [inline osu.Game.Rulesets.Mania/Scoring/ManiaScoreProcessor.cs]: That's fine, HoldNote.Ticks wraps a List and .Count() with on an IEnumerable wrapping a list source is guaranteed to be O(1).

Comment (peppy) [inline osu.Game.Rulesets.Mania/Scoring/ManiaScoreProcessor.cs]: you're inferring a lot there...
you're also only using TotalTicks here. so just calculate it here for now?

File: osu.Game.Rulesets.Mania/Objects/HoldNote.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
 using System.Collections.Generic;
+using System.Linq;
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/Objects/HoldNote.cs]: ?_?

Comment (smoogipooo) [inline osu.Game.Rulesets.Mania/Objects/HoldNote.cs]: Hwoops.

**Review by peppy — APPROVED** [review]

**Pull request merged**